### PR TITLE
feat(min_charge): Create and update a charge with min_amount_cents

### PR DESCRIPTION
**Pull request by rsempe**
## Roadmap Task

👉  https://getlago.canny.io/feature-requests/p/define-minimum-fees

## Context

We want to allow users to set spendings minimum on usage-based charges.

## Description

The goal of this PR is to add the ability to create and update a charge with `min_amount_cents` attribute.

### CODE REVIEW

File: app/services/plans/update_service.rb [hunk 1]
```diff
@@ -81,8 +86,14 @@ def process_charges(plan, params_charges)
           # NOTE: charges cannot be edited if plan is attached to a subscription
           unless plan.attached_to_subscriptions?
             payload_charge[:group_properties]&.map! { |gp| GroupProperty.new(gp) }
+
```
Comment (jdenquin) [inline app/services/plans/update_service.rb:89]: Layout/TrailingWhitespace: Trailing whitespace detected. (https://rubystyle.guide#no-trailing-whitespace)

File: spec/services/plans/update_service_spec.rb [hunk 2]
```diff
@@ -196,7 +196,24 @@
           group_id: group.id,
           values: { 'amount' => '100' },
         )
-        expect(existing_charge).to be_instant
+      end
+
+      it 'does not update premium attributes' do
+        plan = plans_service.call.plan
+
```
Comment (jdenquin) [inline spec/services/plans/update_service_spec.rb:203]: Layout/TrailingWhitespace: Trailing whitespace detected. (https://rubystyle.guide#no-trailing-whitespace)

File: spec/services/plans/update_service_spec.rb [hunk 3]
```diff
@@ -196,7 +196,24 @@
...
+      it 'does not update premium attributes' do
+        plan = plans_service.call.plan
+      
+        expect(existing_charge.reload).not_to be_instant
+        expect(plan.charges.where(instant: false).first.min_amount_cents).to eq(0)
+      end
+
+      context 'when premium' do
+        around { |test| lago_premium!(&test) }
+
```
Comment (jdenquin) [inline spec/services/plans/update_service_spec.rb:210]: Layout/TrailingWhitespace: Trailing whitespace detected. (https://rubystyle.guide#no-trailing-whitespace)

File: spec/services/plans/update_service_spec.rb [hunk 4]
```diff
@@ -196,7 +196,24 @@
...
+        expect(existing_charge.reload).not_to be_instant
+        expect(plan.charges.where(instant: false).first.min_amount_cents).to eq(0)
+      end
+
+      context 'when premium' do
+        around { |test| lago_premium!(&test) }
+  
+        it 'saves premium attributes' do
+          plans_service.call
+
```
Comment (jdenquin) [inline spec/services/plans/update_service_spec.rb:213]: Layout/TrailingWhitespace: Trailing whitespace detected. (https://rubystyle.guide#no-trailing-whitespace)

**Review by lovrocolic — APPROVED** [review]

**Pull request merged**